### PR TITLE
[FEATURE-REQUEST] Import and version checking improvements  #2211

### PR DESCRIPTION
**Pull request by franz101**
Changing to importlib according to this issue:
https://github.com/vaexio/vaex/issues/2211

I was not able to build vex locally yet. Curios about feedback

### CODE REVIEW

File: packages/vaex-core/vaex/version.py [hunk 1]
```diff
@@ -1,15 +1,38 @@
...
     def is_installed(p):
+        #  Concept from https://github.com/geoalchemy/geoalchemy2/pull/392/files#diff-208c493ee84f787a0a966b235a31f86a5d382ff76e1e5533b5b1da712996b7e2
+        try:
+            import importlib.metadata
+        except ImportError:
+            try:
+                from pkg_resources import DistributionNotFound
+                from pkg_resources import get_distribution
+            except ImportError:  # pragma: no cover
+                pass
```
Comment (maartenbreddels) [inline packages/vaex-core/vaex/version.py:14]: why would this import fail?

File: packages/vaex-core/vaex/version.py [hunk 2]
```diff
@@ -1,15 +1,38 @@
...
+            else:
+                try:
+                    return get_distribution(p).version
+                except DistributionNotFound:  # pragma: no cover
+                    raise Exception(f"Package not found: {p}")
```
Comment (maartenbreddels) [inline packages/vaex-core/vaex/version.py:19]: Isn't this much simpler?
```suggestion
                    return False
```

File: packages/vaex-core/vaex/version.py [hunk 3]
```diff
@@ -1,15 +1,38 @@
...
+        else:
+            try:
+                return importlib.metadata.version(p)
+
+            except importlib.metadata.PackageNotFoundError:  # pragma: no cover
+                raise Exception(f"Package not found: {p}")
```
Comment (maartenbreddels) [inline packages/vaex-core/vaex/version.py:25]:
```suggestion
                return False
```

**Review by maartenbreddels — CHANGES_REQUESTED** [review]
Hi,

thanks for you contribution!
I do think the code can use a cleanup, the `is_installed` can simply return a boolean, no need to throw exceptions.
Or, call the function `get_version()`, but I don't like to do string comparisons on exception handling.
I think the cleanest approach is a `is_installed` and `get_version`.
Does that make sense?

Regards,

Maarten

**Comment by franz101** [comment]
I agree with your changes. Currently I found another issue with colab. When you install a package that changes a subdependency it requires  restart. but if you just run the versions command it's fine SMH:
https://colab.research.google.com/drive/1yu0Fkszz7UR5zLYZdWUt_v1net5ewVvQ?usp=sharing

File: packages/vaex-core/vaex/version.py (hunk 1 above)
Comment (franz101) [reply packages/vaex-core/vaex/version.py]: removed

File: packages/vaex-core/vaex/version.py (hunk 2 above)
Comment (franz101) [reply packages/vaex-core/vaex/version.py]: changed the logic to just save the version

**Comment by franz101** [comment]
@maartenbreddels please have a look at the new implementation

**Review by maartenbreddels — APPROVED** [review]
Yeah, i like this a lot!

**Pull request merged**